llvm/utils/TableGen/DXILEmitter.cpp

```diff
@@ -151,7 +151,11 @@ DXILOperationDesc::DXILOperationDesc(const Record *R) {
         assert(knownType && "Specification of multiple differing overload "
                             "parameter types not yet supported");
       } else {
-        OverloadParamIndices.push_back(i);
+        // Skip the return value - nothing is overloaded on only return, and it
```

Comment:
Things like LoadInput is overloaded on only return.

Reply:
That's a good point. This should still happen to work since we treat "not overloaded" and "overloaded on return type" the same for the purposes of overloadKind in tryCreateOp, and I've updated the comment to say so.

The right thing to do, which I think should be done in a follow up, is to generate a table of the function signatures per opcode, which can then be queried by a simple switch over those records. I think we'll need that soon any way so I'll do that shortly.

Reply:
Thinking about this some more I don't think we can get away with this - if an op is overloaded on return type we simply need to provide the one we want. I think I'll add a `Type *` argument that's defaulted to null for "infer it" and return an error if we can't do so.

Reply:
Fixed in latest.